chalice/cli/factory.py

```diff
@@ -157,6 +187,39 @@ def create_log_retriever(self, session, lambda_arn):
         retriever = LogRetriever.create_from_lambda_arn(client, lambda_arn)
         return retriever
 
+    def create_stdin_reader(self):
+        # type: () -> PipeReader
+        stream = click.get_binary_stream('stdin')
+        reader = PipeReader(stream)
+        return reader
+
+    def create_lambda_invoke_handler(self, name, stage):
+        # type: (str, str) -> LambdaInvokeHandler
+        config = self.create_config_obj(stage)
+        deployed = config.deployed_resources(stage)
+        try:
+            resource = deployed.resource_values(name)
+            arn = resource['lambda_arn']
+        except (KeyError, ValueError):
+            raise NoSuchFunctionError(name)
+
+        function_scoped_config = config.scope(stage, name)
+        session = self.create_botocore_session(
+            read_timeout=function_scoped_config.lambda_timeout,
+            max_retries=0,
```

Comment:
Can you add a comment about why we set `max_retries=0`.  It may not be obvious to others why we want to disable retries when invoking the lambda function.